l1-contracts/contracts/state-transition/chain-deps/facets/Mailbox.sol

```diff
@@ -391,14 +401,18 @@ contract MailboxFacet is ZkSyncHyperchainBase, IMailbox {
         bytes32 _canonicalTxHash,
         uint64 _expirationTimestamp
     ) internal {
-        s.priorityQueue.pushBack(
-            PriorityOperation({
-                canonicalTxHash: _canonicalTxHash,
-                // FIXME: safe downcast
-                expirationTimestamp: _expirationTimestamp,
-                layer2Tip: uint192(0) // TODO: Restore after fee modeling will be stable. (SMA-1230)
-            })
-        );
+        if (s.priorityTree.startIndex > s.priorityQueue.getFirstUnprocessedPriorityTx()) {
+            s.priorityQueue.pushBack(
+                PriorityOperation({
+                    canonicalTxHash: _canonicalTxHash,
+                    // FIXME: safe downcast
+                    expirationTimestamp: _expirationTimestamp,
+                    layer2Tip: uint192(0) // TODO: Restore after fee modeling will be stable. (SMA-1230)
+                })
+            );
+        }
+        // TODO: do we have to push keccak256(abi.encode(canonicalTxHash, expirationTimestamp, layer2Tip)) instead?
```

Comment:
Remove this comment. I would propose to keep it as `canonicalTxHash` now and in the future (when escape hatch will be introduced - keccak256(`bytes1(version) || payload`)), where payload depends on the version